spec/Spec/BlsHelpers.cry

```diff
@@ -125,6 +124,79 @@ compute_roots_of_unity order = assert (order == `n) "order must be equal to n" r
         result = compute_powers`{n} root_of_unity'
 
+/**
+ * Bytes used to encode a BLS scalar field element.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#constants
+ */
+type BYTES_PER_FIELD_ELEMENT = 32
+
+/**
+ * The bytes of a BLS scalar field element.
+ */
+type Bytes32 = [BYTES_PER_FIELD_ELEMENT][8]
+
+/**
+ * Convert untrusted bytes to a trusted and validated BLS scalar field element.
+ * This function does not accept inputs greater than the BLS modulus.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#bytes_to_bls_field
+ * ```repl
+ * let bytes = [0x73,0xed,0xa7,0x53,0x29,0x9d,0x7d,0x48,0x33,0x39,0xd8,0x08,0x09,0xa1,0xd8,0x05,0x53,0xbd,0xa4,0x02,0xff,0xfe,0x5b,0xfe,0xff,0xff,0xff,0xff,0x00,0x00,0x00,0x00]
+ * let field_element = 0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000000
+ * :prove bytes_to_bls_field bytes == field_element
+ * ```
+ */
+bytes_to_bls_field : Bytes32 -> BlsFieldElement
+bytes_to_bls_field bytes = assert (field_element < `(BLS_MODULUS)) error_msg field_element where
+    field_element = join bytes // Crytol is big endian
+    error_msg = "BLS scalar field element must be less than modulus"
+
+/**
+ * Perform BLS validation required by the types `KZGProof` and `KZGCommitment`.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#validate_kzg_g1
+ * ```repl
+ * :prove validate_kzg_g1 G1_1P_COMPRESSED
+ * // :error key_validate G1_INVALID_POINT_COMPRESSED // Cryptol cannot currently do this
+ * ```
+ * NOTE: I have manually checked that calling `key_validate` on an invalid compressed point
+ *  fails with an error thrown. Unfortunately, there is not a way in the Cryptol REPL to
+ *  say that I expect an error to be thrown.
+ * NOTE: this repl test is slow.
+ */
+validate_kzg_g1 : Bytes48 -> Bit
+validate_kzg_g1 b =
+    if b == G1_POINT_AT_INFINITY then True
+    else key_validate b
+
+/**
+ * Convert untrusted bytes into a trusted and validated `KZGCommitment`.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#bytes_to_kzg_commitment
+ * ```repl
+ * :prove (bytes_to_kzg_commitment G1_1P_COMPRESSED) == G1_1P_COMPRESSED
+ * ```
+ * NOTE: this repl test is slow.
+ */
+bytes_to_kzg_commitment : Bytes48 -> KZGCommitment
+bytes_to_kzg_commitment b =
+    assert (validate_kzg_g1 b) "validate_kzg_g1 failed" b
+
+/**
+ * Same as for `KZGCommitment`.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#custom-types
+ */
+type KZGProof = Bytes48
+
+/**
+ * Convert untrusted bytes into a trusted and validated `KZGProof`.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#bytes_to_kzg_proof
+  * ```repl
+ * :prove (bytes_to_kzg_commitment G1_1P_COMPRESSED) == G1_1P_COMPRESSED
+ * ```
+ * NOTE: this repl test is slow.
```

Comment:
suggestion: throughout, consider adding a rough time estimate on the "slow" things. I know it might vary by machine but ~10 sec slow is different from ~2 min from ~30 min.

Feel free to skip for now if you don't have all of these handy, but consider doing so in the future.

Reply:
Really good point. Is there a way to capture timing of the tests in Cryptol?

These are on the order of seconds, but it still feels slow compared to almost instantaneous.